samcli/lib/test_runner/test_runner_template_generator.py

```diff
@@ -0,0 +1,347 @@
+import json
+import boto3
+import re
+import logging
+from jinja2 import Template
+from cfn_flip import to_yaml
+from typing import *
+
+LOG = logging.getLogger(__name__)
+
+
+def __extract_api_id_from_arn(api_arn: str) -> str:
```

Comment:
nit: a single underscore prefix is the [more Pythonic](https://peps.python.org/pep-0008/#descriptive-naming-styles) notation for "private" functions